core/src/price_finding.rs

```diff
@@ -1,20 +1,23 @@
+pub mod min_avg_fee;
 pub mod naive_solver;
 pub mod optimization_price_finder;
 pub mod price_finder_interface;
 
-use crate::price_estimation::PriceEstimating;
-pub use crate::price_finding::naive_solver::NaiveSolver;
-pub use crate::price_finding::optimization_price_finder::OptimisationPriceFinder;
-pub use crate::price_finding::price_finder_interface::{
-    Fee, InternalOptimizer, PriceFinding, SolverType,
+pub use self::{
+    min_avg_fee::MinAverageFeeComputing,
+    naive_solver::NaiveSolver,
+    optimization_price_finder::OptimisationPriceFinder,
+    price_finder_interface::{Fee, InternalOptimizer, PriceFinding, SolverType},
 };
+use crate::price_estimation::PriceEstimating;
 use log::info;
+use std::sync::Arc;
 
 pub fn create_price_finder(
     fee: Option<Fee>,
     solver_type: SolverType,
-    price_oracle: impl PriceEstimating + Sync + 'static,
-    min_avg_fee_per_order: u128,
+    price_oracle: Arc<dyn PriceEstimating + Send + Sync>,
+    min_avg_fee: Arc<dyn MinAverageFeeComputing>,
```

Comment:
The Arc is not yet needed, right? Just here for future proof component sharing?

Reply:
It just makes the `MinAverageFeeComputing` to share later, for example if we share it with the solution submission component for economic viability computation.